common/verity.c

```diff
@@ -147,12 +151,133 @@ generate_dm_table_load_extra_params(uint8_t *buf, size_t len, verity_sb_t *sb, c
 char *
 verity_get_device_path_new(const char *label)
 {
-	return mem_printf("%s%s", DM_PATH_PREFIX, label);
+	return mem_printf("%s/%s", DM_PATH_PREFIX, label);
+}
+
+static char *
+verity_get_link_path_new(const char *verity_name)
+{
+	return mem_printf("%s/%s", DM_PATH_PREFIX, verity_name);
+}
+
+static int
+create_dm_symlink(const char *name, const dev_t devt, bool enforce_symlinks)
+{
+	int ret = -1;
+	uevent_event_t *uev = NULL;
+	char *targetpath = NULL, *linkpath = NULL, *buf = NULL;
+	char *uev_path = mem_printf("/sys/dev/block/%lu:%lu/uevent", MAJOR(devt), MINOR(devt));
```

Comment:
don't use the kernel macros here, uses major() minor( ) from sys/sysmacros.h:
https://man7.org/linux/man-pages/man3/makedev.3.html